src/main/java/com/amam/wizardschool/controller/FacultyController.java

```diff
@@ -52,8 +59,26 @@ public ResponseEntity<Faculty> deleteFaculty(@PathVariable("id") Long id) {
         }
     }
 
-    @GetMapping
+    @GetMapping("/color")
+    @Operation(summary = "Get faculties by color")
     public ResponseEntity<Collection<Faculty>> getFacultyByColor(@RequestParam("color") String color) {
         return ResponseEntity.ok(facultyService.getFacultyByColor(color));
     }
+
+    @GetMapping
+    @Operation(summary = "Get faculty by Name or Color")
+    public ResponseEntity<Collection<Faculty>> getFacultyByNameOrColorIgnoreCase(@RequestParam(required = false) String name,
+                                                                                 @RequestParam(required = false) String color) {
+        if (name == null && color == null) {
+            return ResponseEntity.badRequest().build();
+        }
+        return ResponseEntity.ok(facultyService.getFacultyByNameOrColorIgnoreCase(name, color));
+    }
+
+    @GetMapping("/faculty")
```

Comment:
здесь ошибка REST тоже - путь должен быть с id, потому что идёт обращение к ресурсу факультета по id и у него уже берутся студенты
```suggestion
    @GetMapping("/{id}/students")
```